tests/nested/manual/optee-fde/task.yaml

```diff
@@ -0,0 +1,200 @@
+    NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL: http://localhost:11028
+
+    KEY_PROTECTOR/optee: "optee"
+    KEY_PROTECTOR/hooks: "fde-setup-hook"
+
+prepare: |
+    if [ "${TRUST_TEST_KEYS}" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    kernel_name=""
+    kernel_revision=""
+    gadget_name=""
+    gadget_revision=""
+    if [ "${KEY_PROTECTOR}" = "optee" ]; then
+        kernel_name="test-qemu-optee-kernel"
+        kernel_revision=2
+        gadget_name="test-qemu-optee-gadget"
+        gadget_revision=5
+    else
+        kernel_name="test-qemu-fde-hooks-kernel"
+        kernel_revision=2
+        gadget_name="test-qemu-fde-hooks-gadget"
+        gadget_revision=3
+    fi
+
+    tests.pkgs install u-boot-tools
+
+    # these snaps are originally built from snaps that live in the
+    # github.com/canonical/optee-uc-fde repo. building the snaps here in CI
+    # would be more ideal, but that would take a really long time.
+    #
+    # specific revisions are used to help allow development on the snaps without
+    # breaking current/old tests.
+
+    snap download --revision "${kernel_revision}" --basename "${kernel_name}" "${kernel_name}"
+    snap download --revision "${gadget_revision}" --basename "${gadget_name}" "${gadget_name}"
+    snap download --edge --basename core24 core24
+
+    unsquashfs -d "./${kernel_name}" "${kernel_name}.snap"
+
+    # TODO: update this to use something like uc24_build_initramfs_kernel_snap,
+    # but for arm
+    #
+    # repack the FIT image in the kernel snap so that we can use the
+    # snap-bootstrap from this branch
+    mkdir ./repack
+    pushd ./repack
+    dumpimage -T flat_dt -p0 -o Image "../${kernel_name}/kernel.img"
+    dumpimage -T flat_dt -p1 -o initrd.img "../${kernel_name}/kernel.img"
+
+    mkdir ./initrd
+    pushd ./initrd
+
+    unzstd < ../initrd.img | cpio -idmv
+    cp -a /usr/lib/snapd/snap-bootstrap ./lib/snapd/snap-bootstrap
+    find . -print0 | cpio --null --create --quiet --format=newc --owner=0:0 | zstd > ../initrd.img
+
+    popd # ./initrd
+    rm -rf ./initrd
+
+    # TODO: what keys should we actually be using here? firmware that lives in
+    # the gadget currently expects these keys
+    git clone https://git.launchpad.net/~ondrak/+git/dev-keys
+
+    cp "../${kernel_name}/qemu.its" ./qemu.its
+    mkimage --fit ./qemu.its -k ./dev-keys ./kernel.img
+
+    popd # ./repack
+
+    mv ./repack/kernel.img "./${kernel_name}/kernel.img"
+    rm -rf ./repack
+
+    snap pack --filename "./${kernel_name}.snap" "./${kernel_name}"
+
+    # although nested_start_core_vm_unit usually does this, the fake store will
+    # already have been set up, so we need to install these snaps here
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    # some of the scripts install this stuff (even though we won't use it on
+    # arm) and our usage of the fake store makes it fail
```

Comment:
"even though we won't use it arm" -> not totally true, right? We just do not use it in this test.